docker/Dockerfile

```diff
@@ -42,19 +39,23 @@ RUN rm /tmp/search-guard-6-6.8.6-25.5.zip
 # Add default roles
 ADD sg_roles_mapping.yml /usr/share/elasticsearch/plugins/search-guard-6/sgconfig/sg_roles_mapping.yml
 
+# Allow read-only anonymous access
+RUN sed -i "s/anonymous_auth_enabled: false/anonymous_auth_enabled: true/" /elasticsearch/plugins/search-guard-6/sgconfig/sg_config.yml
+
 # Change log level INFO->DEBUG
 
 #RUN TMP_FILE=`mktemp /tmp/config.XXXXXXXXXX` && \
 #    sed -e "s/INFO/INFO/" /elasticsearch/config/logging.yml > $TMP_FILE && \
 #    mv $TMP_FILE /elasticsearch/config/logging.yml
 
-# Define working directory.
-WORKDIR /data
+# Define mountable directories.
+VOLUME ["/elasticsearch/data"]
```

Comment:
You are right adding this path as a volume, but it should be added in a different place, so the mount point can be created and chown-ed before the `VOLUME` instruction is set. Something like this:

```
RUN adduser -D -u 1000 esuser
RUN mkdir /elasticsearch/data
RUN chown -R esuser /elasticsearch

# Define mountable directories.
VOLUME ["/elasticsearch/data"]
```

Reply:
The reason I've done it this way (with a `chown` as part of the container init in `docker-entrypoint.sh`), is because I believe it's the only way to work around permissions if you're launching a container from this image and mapping a folder from the host.

My use-case is that I'm running GrimoireLab on DC/OS and I need to be able to mount folders from the container's host for data persistence.

Reply:
Hmmm, what's the id of the user executing the docker-compose? 

When we deploy in cloud servers we create a non-system user (uid 1000) which launches the compose file. So the only thing we have to do is to make sure the directories to be mounted as bind volumes exist and are owned by this user. I know it is not a big difference, but I don't like the idea of allowing the container to change the permission of content owned by root and give it to user with uid 1000.

Reply:
You're fine as long as the UID on the host matches the UID in the container, but otherwise it's a [long-standing](https://github.com/moby/moby/issues/7198) source of pain with Docker containers.  It also doesn't matter if you're not mapping folders from the host into the container when using docker-compose.

The issue I've linked to mentions [fixuid](https://github.com/boxboat/fixuid) as one possible workaround / solution, but I haven't used it before.